src/libraries/System.Runtime.InteropServices/gen/ComInterfaceGenerator/Marshallers/ComInterfaceDispatchMarshallerFactory.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using static Microsoft.CodeAnalysis.CSharp.SyntaxFactory;
+
+namespace Microsoft.Interop
+{
+    internal sealed record ComInterfaceDispatchMarshallingInfo : MarshallingInfo
+    {
+        public static readonly ComInterfaceDispatchMarshallingInfo Instance = new();
+    }
+
+    internal sealed class ComInterfaceDispatchMarshallerFactory : IMarshallingGeneratorFactory
+    {
+        private readonly IMarshallingGeneratorFactory _inner;
+        public ComInterfaceDispatchMarshallerFactory(IMarshallingGeneratorFactory inner)
+        {
+            _inner = inner;
+        }
+
+        public IMarshallingGenerator Create(TypePositionInfo info, StubCodeContext context)
+            => info.MarshallingAttributeInfo is ComInterfaceDispatchMarshallingInfo ? new Marshaller() : _inner.Create(info, context);
+
+        private sealed class Marshaller : IMarshallingGenerator
+        {
+            public ManagedTypeInfo AsNativeType(TypePositionInfo info) =>
+                new PointerTypeInfo(
+                    $"{TypeNames.System_Runtime_InteropServices_ComWrappers_ComInterfaceDispatch}*",
+                    $"{TypeNames.System_Runtime_InteropServices_ComWrappers_ComInterfaceDispatch}*",
```

Comment:
I can't recall, but are these supposed to be the same? I thought one was "to" and the other was "from".

Reply:
The names here are the "full name" and the "diagnostic full name" for reporting in diagnostics to the user. They're the same in most cases.